cosmwasm/packages/wasmi-runtime/src/exports.rs

```diff
@@ -124,17 +155,27 @@ pub unsafe extern "C" fn ecall_handle(
     let env = std::slice::from_raw_parts(env, env_len);
     let msg = std::slice::from_raw_parts(msg, msg_len);
     let result = panic::catch_unwind(|| {
-        let used_gas_ref = &mut *used_gas;
-        let result = crate::wasm::handle(context, gas_limit, used_gas_ref, contract, env, msg);
+        let mut local_used_gas = *used_gas;
```

Comment:
Can you add a comment why you did it this way?

Reply:
Both works, but @reuvenpo and I thought that it improves readability.
It was hard figuring where these references came from while chasing gas issues.